code/modules/mob/living/basic/icemoon/ice_whelp/ice_whelp_ai.dm

```diff
@@ -0,0 +1,217 @@
+		if(!istype(potential_food, living_pawn.type))
+			continue
+		if(potential_food.stat != DEAD)
+			continue
+		if(potential_food.pulledby) //someone already got him before us
+			continue
+		to_be_consumed = potential_food
+		break
+
+	if(!to_be_consumed)
+		finish_action(controller, FALSE)
+		return
+
+	controller.set_blackboard_key(target_key, to_be_consumed)
+	finish_action(controller, TRUE)
+
+/datum/ai_behavior/cannibalize
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_REQUIRE_REACH | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION
+	///the hp we'll regain after cannibalizing
+	var/health_boost = 5
+
+/datum/ai_behavior/cannibalize/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/atom/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/cannibalize/perform(seconds_per_tick, datum/ai_controller/controller, target_key, attack_key)
+	. = ..()
+	var/mob/living/living_pawn = controller.pawn
+	var/mob/living/target = controller.blackboard[target_key]
+
+	if(QDELETED(target))
+		finish_action(controller, FALSE)
+		return
+
+	if(target.stat != DEAD || target.pulledby) //we were too slow
+		finish_action(controller, FALSE)
+		return
+
+	living_pawn.start_pulling(target)
+	if(!do_after(living_pawn, 5 SECONDS, target))
+		finish_action(controller, FALSE)
+		return
+
+	target.gib()
+	living_pawn.adjustBruteLoss(-1 * health_boost)
+	finish_action(controller, TRUE)
+
+/datum/ai_behavior/cannibalize/finish_action(datum/ai_controller/controller, succeeded, target_key)
+	. = ..()
+	controller.clear_blackboard_key(target_key)
+
+///subtree to find icy rocks and create sculptures out of them
+/datum/ai_planning_subtree/sculpt_statues
+
+/datum/ai_planning_subtree/sculpt_statues/SelectBehaviors(datum/ai_controller/controller, seconds_per_tick)
+	var/obj/target = controller.blackboard[BB_TARGET_ROCK]
+
+	if(QDELETED(target))
+		controller.queue_behavior(/datum/ai_behavior/find_and_set, BB_TARGET_ROCK, /obj/structure/flora/rock/icy)
+		return
+
+	controller.queue_behavior(/datum/ai_behavior/sculpt_statue, BB_TARGET_ROCK)
+	return SUBTREE_RETURN_FINISH_PLANNING
+
+/datum/ai_behavior/sculpt_statue
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_REQUIRE_REACH | AI_BEHAVIOR_CAN_PLAN_DURING_EXECUTION
+	action_cooldown = 5 MINUTES
+
+/datum/ai_behavior/sculpt_statue/setup(datum/ai_controller/controller, target_key)
+	. = ..()
+	var/obj/target = controller.blackboard[target_key]
+	if(QDELETED(target))
+		return FALSE
+	set_movement_target(controller, target)
+
+/datum/ai_behavior/sculpt_statue/perform(seconds_per_tick, datum/ai_controller/controller, target_key)
+	. = ..()
+
+	var/atom/target = controller.blackboard[target_key]
+	var/mob/living/basic/mining/ice_whelp/living_pawn = controller.pawn
+
+	if(QDELETED(target))
+		finish_action(controller, FALSE, target_key)
+		return
+
+	living_pawn.create_sculpture(target)
```

Comment:
This behavior should just, `ClickOn` the target, to which the mob itself will handle converting click -> attack -> create scultpure